tests/unit/models/builder-test.js

```diff
@@ -293,6 +597,94 @@ describe('models/builder.js', function() {
       buildResults = 'build results';
     });
 
+    describe('instrumentation', function() {
+      beforeEach(function() {
+        return mkTmpDirIn(tmproot).then(function (dir) {
+          process.chdir(dir);
+        });
+      });
+
+      afterEach(function() {
+        process.chdir(root);
+        return remove(tmproot);
+      });
+
+      it('invokes build instrumentation hook when EMBER_CLI_INSTRUMENTATION=1', function() {
+        process.env.EMBER_CLI_INSTRUMENTATION = '1';
+        var mockVizInfo = Object.create(null);
+        var mockResultAnnotation = Object.create(null);
+
+        builder._computeVizInfo = function (count, result, resultAnnotation) {
+          expect(count).to.be.gte(0);
+          expect(result).to.equal(buildResults);
+          expect(resultAnnotation).to.equal(mockResultAnnotation);
+          return mockVizInfo;
+        }
+
+        return builder.build('stringdir', mockResultAnnotation).then(function() {
+          expect(hooksCalled).to.include('buildInstrumentation');
+          expect(instrumentationArg).to.equal(mockVizInfo);
+        });
+      });
+
+      it('writes and invokes build instrumentation hook when BROCCOLI_VIZ=1', function() {
+        process.env.BROCCOLI_VIZ = '1';
+        var mockVizInfo = Object.create(null);
+        var mockResultAnnotation = Object.create(null);
+
+        builder._computeVizInfo = function (count, result, resultAnnotation) {
```

Comment:
I think we should use test double here:

how to use test double for this: https://github.com/ember-cli/ember-cli/pull/6461#discussion_r92225899

(future, we may want to document this in contributing.md)